gc/base/MemorySubSpace.cpp

```diff
@@ -919,16 +919,23 @@ MM_MemorySubSpace::systemGarbageCollect(MM_EnvironmentBase* env, uint32_t gcCode
 	}
 
 	if (_collector && _usesGlobalCollector) {
-		/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
-		_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
-
 		env->acquireExclusiveVMAccessForGC(_collector);
-		reportSystemGCStart(env, gcCode);
-
-		/* system GCs are accounted into "user" time in GC/total time ratio calculation */
-		_collector->garbageCollect(env, this, NULL, gcCode, NULL, NULL, NULL);
-
-		reportSystemGCEnd(env);
+		#if defined(OMR_GC_IDLE_HEAP_MANAGER)
+		if (isSystemGCAllowed(env, gcCode)) {
+		#endif
+			/* TODO: This is bogus for multiple memory spaces - should ask the space, not the heap */
+			_extensions->heap->getResizeStats()->setFreeBytesAtSystemGCStart(getApproximateActiveFreeMemorySize());
```

Comment:
Ah I see the answer to my question... releaseFreeMemoryPages is invoked just after. 

I'll think a bit what's the best way... But for now, I'll say this: walking&decommit a pool should be safe by just holding tenure allocate lock - exclusive access is not really needed. So, one way to do it is 1) to pull out the 'is allowed' logic out of sysGC call. 2) if sysGC call is done, invoke releaseFreeMemoryPages somwhere in Global GC code, toward the end. 3) invoke releaseFreeMemoryPages  in else clause of 'is allowed', outside of sysGC, but add pool locking the the release logic. Dmitiri would you find this resonable?